src/Block/SharedBlockBlockService.php

```diff
@@ -134,18 +135,17 @@ public function load(BlockInterface $block): void
         $block->setSetting('blockId', $sharedBlock);
     }
 
-    private function getBlockBuilder(): FormBuilderInterface
+    private function getBlockBuilder(FormMapper $form): FormBuilderInterface
     {
         $fieldDescription = $this->sharedBlockAdmin->createFieldDescription('block', [
             'translation_domain' => 'SonataPageBundle',
             'edit' => 'list',
         ]);
         $fieldDescription->setAssociationAdmin($this->sharedBlockAdmin);
 
-        return $this->sharedBlockAdmin->getFormContractor()->getFormBuilder(
-            $this->sharedBlockAdmin->getUniqId(),
-            ['data_class' => $this->sharedBlockAdmin->getClass()]
-        )->create('blockId', ModelListType::class, [
+        \assert($form instanceof AdminFormMapper);
```

Comment:
I don't like the assert here. 

Nothing is preventing
```
configureEditForm
```
to be called with another FormMapper.

I would change this too
```
private function getBlockBuilder(AdminFormMapper $form): FormBuilderInterface
```
and
```
public function configureEditForm(FormMapper $form, BlockInterface $block): void
    {
        if (!$form instanceof AdminFormMapper) {
             throw new \InvalidArgumentException();
        }
```
(And we can consider later to make the EditableBlockService generics)

Reply:
What do you think now?